modules/layers/src/solid-polygon-layer/solid-polygon-layer.js

```diff
@@ -80,7 +80,7 @@ export default class SolidPolygonLayer extends Layer {
     this.setState({
       numInstances: 0,
       polygonTesselator: new PolygonTesselator({
-        preproject: coordinateSystem === COORDINATE_SYSTEM.LNGLAT,
+        preproject: coordinateSystem === COORDINATE_SYSTEM.LNGLAT && viewport.projectFlat,
```

Comment:
Nits: 
- Add comment please, explaining the condition.
- Not super happy with the condition here scraping for the presence of a method, it feels a little  sloppy in something that is getting quite complicated. Could we add a semantic flag to the viewport that we can check (`supportsFlatProjection()`) so that this seems like a more defined interface, or improve in some other way?

Reply:
`projectFlat` is a `Viewport` base class methods. This is the same as

```js
coordinateSystem === COORDINATE_SYSTEM.LNGLAT ? viewport.projectFlat : false
```